californium-core/src/main/java/org/eclipse/californium/core/network/stack/BlockwiseLayer.java

```diff
@@ -716,10 +764,21 @@ private void handleBlock1Response(final Exchange exchange, final Response respon
 			Block1BlockwiseStatus status = getBlock1Status(key);
 
 			if (status == null) {
-
-				// request has not been sent blockwise
-				LOGGER.debug("discarding unexpected block1 response: {}", response);
-
+				if (enableAutoFailoverOn413 && response.getCode() == ResponseCode.REQUEST_ENTITY_TOO_LARGE) {
+					Request request = exchange.getRequest();
+					if (block1.getNum() == 0 && block1.getSize() < request.getPayloadSize()) {
+						// Start block1 transfer
+						Request blockRequest = startBlockwiseUpload(exchange, request,
+								Math.min(block1.getSize(), preferredBlockSize));
+						exchange.setCurrentRequest(blockRequest);
+						lower().sendRequest(exchange, blockRequest);
+					} else {
+						deliverResponseToUpperLayer(exchange, response);
+					}
+				} else {
+					// request has not been sent blockwise
+					LOGGER.debug("discarding unexpected block1 response: {}", response);
+				}
```

Comment:
I would prefer, if the failover already creates the Block1BlockwiseStatus, so that the null case could be kept simple here.